File: yemeksiparismete_mobile/lib/models/product.dart
class Product {
  final int id;
  final int restaurantId;
  final String name;
  final double price;
  final String description;
  final int stock;
  final String? imageUrl;
  final String category;

  Product({
    required this.id,
    required this.restaurantId,
    required this.name,
    required this.price,
    required this.description,
    required this.stock,
    this.imageUrl,
    required this.category,
  });

  factory Product.fromJson(Map<String, dynamic> json) {
    return Product(
      id: json['id'] ?? 0,
      restaurantId: json['restaurantId'] ?? 0,
      name: json['name'] ?? '',
      price: (json['price'] ?? 0.0).toDouble(),
      description: json['description'] ?? '',
      stock: json['stock'] ?? 0,
      imageUrl: json['imageUrl'],
      category: json['category'] ?? '',
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'restaurantId': restaurantId,
      'name': name,
      'price': price,
      'description': description,
      'stock': stock,
      'imageUrl': imageUrl,
      'category': category,
    };
  }
}

File: yemeksiparismete.Server/Models/Product.cs
namespace yemeksiparismete.Server.Models
{
    public class Product
    {
        public int Id { get; set; }
        public int RestaurantId { get; set; }
        public string Name { get; set; } = string.Empty;
        public decimal Price { get; set; }
        public string Description { get; set; } = string.Empty;
        public int Stock { get; set; }
        public string ImageUrl { get; set; } = string.Empty;
        public string Category { get; set; } = string.Empty;
    }
}
